sklearn/preprocessing/_discretization.py

```diff
@@ -233,9 +241,18 @@ def fit(self, X, y=None):
                 '`subsample` must be used with `strategy="quantile"`.'
             )
 
+        elif sample_weight is not None and self.strategy != "quantile":
+            raise ValueError(
+                "`sample_weight` was provided but it can be only used with"
+                f"strategy='quantile'. Got strategy={self.strategy!r} instead."
```

Comment:
```suggestion
                "`sample_weight` was provided but it can only be "
                f"used with strategy='quantile'. Got strategy="
                f"{self.strategy!r} instead."
```